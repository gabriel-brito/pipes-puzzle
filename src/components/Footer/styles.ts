import styled from 'styled-components'

export const Wrapper = styled.footer`
  align-items: center;
  background-color: var(--bgContrast);
  border-radius: 8px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 0 auto 32px;
  padding: 16px;
  width: fit-content;
`

export const Information = styled.h2`
  font-size: 16px;
  margin: 0 auto 16px;
  text-align: center;
`

export const Button = styled.button`
  background-color: var(--buttonBg);
  background: linear-gradient(135deg, #ff80bf 0, #9580ff 100%);
  border-radius: 8px;
  border: none;
  color: var(--buttonColor);
  cursor: pointer;
  font-size: 16px;
  font-size: 600;
  font-weight: 900;
  height: 48px;
  width: 200px;

  &:hover {
    opacity: 0.7;
  }
`
