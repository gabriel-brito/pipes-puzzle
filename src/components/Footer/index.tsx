import * as S from './styles'

type FooterTypes = {
  handleNextLevel: () => void
  handleVerify: () => void
  isNextLevelAllowed: boolean
  resultMessage?: string
}

export default function Footer({
  handleNextLevel,
  handleVerify,
  isNextLevelAllowed,
  resultMessage
}: FooterTypes) {
  return (
    <S.Wrapper data-testid="footer">
      {resultMessage && <S.Information>{resultMessage}</S.Information>}

      {!isNextLevelAllowed ? (
        <S.Button onClick={handleVerify} aria-label="verify-solution-button">
          Verify Solution
        </S.Button>
      ) : (
        <S.Button onClick={handleNextLevel} aria-label="next-level-button">
          Next Level
        </S.Button>
      )}
    </S.Wrapper>
  )
}
