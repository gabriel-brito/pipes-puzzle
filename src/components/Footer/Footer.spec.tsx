import { render, screen } from '@testing-library/react'
import UserEvent from '@testing-library/user-event'
import 'jest-styled-components'

import Footer from '.'

const props = {
  handleNextLevel: jest.fn(),
  handleVerify: jest.fn(),
  isNextLevelAllowed: false,
  resultMessage: ''
}

describe('Footer', () => {
  it('should render properly', () => {
    render(<Footer {...props} />)

    expect(
      screen.getByRole('button', { name: /verify-solution-button/i })
    ).toBeInTheDocument()
    expect(screen.getByTestId('footer')).toMatchSnapshot()
  })

  it('should call the handleVerify func when isNextLevelAllowed is false', () => {
    render(<Footer {...props} />)

    const button = screen.getByRole('button', {
      name: /verify-solution-button/i
    })

    UserEvent.click(button)

    expect(props.handleVerify).toHaveBeenCalledTimes(1)
  })

  it('should call the handleNextLevel func when isNextLevelAllowed is true', () => {
    render(<Footer {...props} isNextLevelAllowed={true} />)

    const button = screen.getByRole('button', {
      name: /next-level-button/i
    })

    UserEvent.click(button)

    expect(props.handleNextLevel).toHaveBeenCalledTimes(1)
  })

  it('should show the result message', () => {
    render(<Footer {...props} resultMessage="Correct!" />)

    expect(
      screen.getByRole('heading', { name: /Correct/i })
    ).toBeInTheDocument()
  })
})
