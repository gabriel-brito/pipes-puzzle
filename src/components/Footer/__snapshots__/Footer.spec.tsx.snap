// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render properly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: var(--bgContrast);
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto 32px;
  padding: 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c1 {
  background-color: var(--buttonBg);
  background: linear-gradient(135deg,#ff80bf 0,#9580ff 100%);
  border-radius: 8px;
  border: none;
  color: var(--buttonColor);
  cursor: pointer;
  font-size: 16px;
  font-size: 600;
  font-weight: 900;
  height: 48px;
  width: 200px;
}

.c1:hover {
  opacity: 0.7;
}

<footer
  class="c0"
  data-testid="footer"
>
  <button
    aria-label="verify-solution-button"
    class="c1"
  >
    Verify Solution
  </button>
</footer>
`;
